pkg/flow/internal/controller/loader.go

```diff
@@ -468,7 +489,14 @@ func (l *Loader) Variables() map[string]interface{} {
 func (l *Loader) Components() []*ComponentNode {
 	l.mut.RLock()
 	defer l.mut.RUnlock()
-	return l.components
+	return l.componentNodes
+}
+
+// Services returns the current set of service nodes.
+func (l *Loader) Services() []*ServiceNode {
```

Comment:
The mutexes are critical because it protects the slice field of `Loader`, not the data inside the slices. 

Without the mutexes, it's possible to get a corrupted slice pointer if the field is updated in parallel with a call to Services and Components.

Reply:
I see thank you. If we don't need to protect the data inside the slices, then i guess that's ok.